Sources/DeepLearning/Initializers.swift

```diff
@@ -13,9 +13,295 @@
+    ///   tensor attribute from `convolved2D`'s `strides` argument.
+    @inlinable
+    internal init(_ scalars: (Scalar, Scalar, Scalar, Scalar)) {
+        self.init([scalars.0, scalars.1, scalars.2, scalars.3])
+    }
+}
+
+internal extension Tensor where Scalar : TensorFlowFloatingPoint {
+    @inlinable
+    static func _vjpInit(
+        repeating repeatedValue: Scalar,
+        shape: TensorShape
+    ) -> (Tensor, (Tensor) -> Scalar) {
+        return (Tensor(repeating: repeatedValue, shape: shape), {
+            $0.sum().scalarized()
+        })
+    }
+}
+
+//===------------------------------------------------------------------------------------------===//
+// Casting
+//===------------------------------------------------------------------------------------------===//
+
+public extension Tensor where Scalar : Numeric {
+    /// Perform an element-wise type conversion from a `Bool` tensor.
+    @inlinable
+    init(_ other: Tensor<Bool>) {
+        self = Raw.cast(other)
+    }
+
+    /// Perform an element-wise conversion from another `Tensor`.
+    @inlinable
+    @differentiable(
+        vjp: _vjpCast where Scalar : TensorFlowFloatingPoint,
+                            OtherScalar: TensorFlowFloatingPoint)
+    init<OtherScalar : Numeric>(_ other: Tensor<OtherScalar>) {
+        self = Raw.cast(other)
+    }
+}
+
+internal extension Tensor where Scalar : TensorFlowFloatingPoint {
+    @inlinable
+    static func _vjpCast<OtherScalar : TensorFlowFloatingPoint>(
+        _ other: Tensor<OtherScalar>
+    ) -> (Tensor, (Tensor) -> Tensor<OtherScalar>) {
+        return (Tensor(other), { v in Tensor<OtherScalar>(v) })
+    }
+}
+
+//===------------------------------------------------------------------------------------------===//
+// Stacking / Concatenating / Tiling
+//===------------------------------------------------------------------------------------------===//
+
+public extension Tensor {
+    /// Creates a tensor from an array of tensors (which may themselves be scalars).
+    @inlinable
+    // @differentiable(where Scalar : TensorFlowFloatingPoint)
+    init(_ elements: [Tensor]) {
+        self = Tensor(stacking: elements)
+    }
+
+    /// Stacks `tensors`, along the `axis` dimension, into a new tensor with rank one higher than
+    /// the current tensor and each tensor in `tensors`.
+    /// 
+    /// Given that `tensors` all have shape `[A, B, C]`, and `tensors.count = N`, then:
+    /// - if `axis == 0` then the resulting tensor will have the shape `[N, A, B, C]`.
+    /// - if `axis == 1` then the resulting tensor will have the shape `[A, N, B, C]`.
+    /// - etc.
+    ///
+    /// For example:
+    /// ```
+    /// // 'x' is [1, 4]
+    /// // 'y' is [2, 5]
+    /// // 'z' is [3, 6]
+    /// Tensor(stacking: [x, y, z]) // is [[1, 4], [2, 5], [3, 6]]
+    /// Tensor(stacking: [x, y, z], alongAxis: 1) // is [[1, 2, 3], [4, 5, 6]]
+    /// ```
+    ///
+    /// This is the opposite of `Tensor.unstacked`.
```

Comment:
Thanks a lot for the detailed doc comments! Since this comment mentions `unstacked`, would you like to include it in this PR as well? Also, what's the type signature of it?